app/api.js

```diff
@@ -52,72 +52,121 @@ export function initIPFSClient () {
 }
 
 /**
- * This function will allow the user to add a file to the IPFS repo.
+ * ```
+ * Link {
+ *  hash: string;
+ *  path: string;
+ *  size: number;
+ * }
+ * ```
+ *
+ * @param {Link[]} links
+ * @returns {Link} the wrapper, which is of type Link
+ */
+export function wrapFiles (links) {
+  const wrapperDAG = {
+    Data: Buffer.from('\u0008\u0001'),
+    // The object.put API expects `Name` not `path`
+    Links: links.map(link => ({
+      Name: link.path,
+      Hash: link.hash,
+      Size: link.size
+    }))
+  }
+
+  return IPFS_CLIENT.object.put(wrapperDAG)
+    .then(res => {
+      // res is of type DAGNode
+      // https://github.com/ipld/js-ipld-dag-pb#nodetojson
+      res = res.toJSON()
+      const wrapper = {
+        hash: res.multihash,
+        path: '',
+        size: res.size
+      }
+
+      return wrapper
+    })
+}
+
+/**
+ * This function will allow the user to add a file or multiple files to the IPFS repo.
+ * Accepts a string or a string array.
+ * Wraps the files in a directory.
+ *
+ * ```
+ * Wrapper {
+ *  hash: string;
+ *  path: string;
+ *  size: number;
+ * }
+ * ```
+ *
+ * @param {string|string[]} filePath
+ * @returns {Wrapper} wrapper
  */
-export function addFileFromFSPath (filePath, _queryGateways = queryGateways) {
+export function addFileOrFilesFromFSPath (filePath, _queryGateways = queryGateways) {
```

Comment:
It will simplify a lot if you could switch to always array (you can have an array of 1 element).
This will remove some conditions in the code and simplify the logic, but it will require on the implementation of the function to use arrays.

Don't mix types of variables

Reply:
https://github.com/Siderus/Orion/pull/95/files#diff-45bba5770252947b22079e6d12f321dfR40
Is it returning a promise?